### PR TITLE
Refactor adapters

### PR DESCRIPTION
**Pull request by scarroll32**
Closes https://github.com/activerecord-hackery/ransack/issues/1346

### CODE REVIEW

**Comment by scarroll32** [comment]
@deivid-rodriguez could you please review this refactoring. It should go after https://github.com/activerecord-hackery/ransack/pull/1345

I had intended to completely remove the Ransack::Adapter::ActiveRecord namespace and replace it with `Ransack::Adapter::ActiveRecord` but I could not get past a connection error. It can be a follow-up issue.

**Comment by scarroll32** [comment]
@deivid-rodriguez can we get this into the 4.0.0 release also?

**Comment by deivid-rodriguez** [comment]
I can try to review it, can you fix conflicts?

**Comment by scarroll32** [comment]
> I can try to review it, can you fix conflicts?

@deivid-rodriguez sorry I didn't realise it had conflicts, will fix asap.

**Comment by scarroll32** [comment]
It's ready for review now @deivid-rodriguez

File: lib/polyamorous.rb [hunk 1]
```diff
@@ -1 +0,0 @@
-require 'polyamorous/polyamorous'
```
[the remark this answers is not in the document]
Comment (scarroll32) [reply lib/polyamorous.rb:1]: @deivid-rodriguez I found this line to be unnecessary, but you are right: it is unrelated to the adapter refactoring. I will pull it into it's own PR.

File: lib/ransack/nodes/condition.rb [hunk 2]
```diff
@@ -2,8 +2,8 @@ module Ransack
   module Nodes
     class Condition < Node
       i18n_word :attribute, :predicate, :combinator, :value
-      i18n_alias :a => :attribute, :p => :predicate,
-                 :m => :combinator, :v => :value
+      i18n_alias a: :attribute, p: :predicate,
+                 m: :combinator, v: :value
```
[the remark this answers is not in the document]
Comment (scarroll32) [reply lib/ransack/nodes/condition.rb:6]: Good point @deivid-rodriguez, I'll pull this into it's own PR.

File: lib/ransack.rb [hunk 3]
```diff
@@ -21,7 +20,11 @@ class UntraversableAssociationError < StandardError; end
 require 'ransack/ransacker'
 require 'ransack/translate'
 
-Ransack::Adapters.object_mapper.require_adapter
+require 'ransack/adapters/active_record/ransack/translate'
+require 'ransack/adapters/active_record'
+require 'ransack/adapters/active_record/ransack/visitor'
+require 'ransack/adapters/active_record/ransack/nodes/condition'
+require 'ransack/adapters/active_record/ransack/context'
```
[the remark this answers is not in the document]
Comment (scarroll32) [reply lib/ransack.rb]: I'll check this and get back to you @deivid-rodriguez

**Comment by scarroll32** [comment]
@deivid-rodriguez I will look at these comments over the weekend. I wonder if we should hold the 4.0.0 release so that this can come in?

It is not a breaking change, but it is a large refactoring. I can go either way (either 4.0.0 or 4.1.0) WDYT ? 🤔

**Comment by deivid-rodriguez** [comment]
I think we can hold it until you can have a look. Breaking `require "polyamorous"` could be considered a breaking change so I'd ship it with 4.0.0 too.

Thanks for this @scarroll32!

File: lib/ransack.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (scarroll32) [reply lib/ransack.rb]: > But I guess we have enough with ActiveRecord?

@deivid-rodriguez I think we keep it to ActiveRecord only, we already removed MongoDB.

**Comment by scarroll32** [comment]
@deivid-rodriguez could you please take another look at this in light of https://github.com/activerecord-hackery/ransack/pull/1371 and https://github.com/activerecord-hackery/ransack/pull/1370 ?

**Review by deivid-rodriguez — APPROVED** [review]
@scarroll32 Yeah, sure. I haven't suffered the pain from maintaining these adapters, but if this is painful and no other ORM framework is supported other than ActiveRecord anyways, then it makes sense to get rid of all this.

**Comment by gregmolnar** [comment]
This was a pretty bad decision. Even if someone would want to maintain the mongoid adapter, or create another one, now it is impossible.

**Pull request merged**